app/src/main/java/com/aptoide/uploader/account/AutoLoginManager.java

```diff
@@ -6,56 +6,113 @@
 import io.reactivex.Single;
 
 public class AutoLoginManager {
-  private Context context;
-  private AutoLoginPersistence persistence;
+  private static Context context;
+  private AutoLoginCredentials autoLoginCredentials = new AutoLoginCredentials();
+  private static AutoLoginManager single_instance;
 
-  public AutoLoginManager(Context context, AutoLoginPersistence persistence) {
+  public AutoLoginManager(Context context) {
     this.context = context;
-    this.persistence = persistence;
+  }
+
+  public static AutoLoginManager getInstance(Context context) {
+    if (single_instance == null) single_instance = new AutoLoginManager(context);
+    return single_instance;
+  }
+
+  public AutoLoginCredentials getAutoLoginCredentials() {
+    return autoLoginCredentials;
   }
 
   public Single<AutoLoginCredentials> getStoredUserCredentials() {
+    return Single.just(autoLoginCredentials);
+  }
+
+  public Single<AutoLoginCredentials> getFirstStoredUserCredentials() {
```

Comment:
I agree. I'm not sure, but actually i believe that  I was the one who added the first one. The reasoning behind it was to avoid going to fetch Vanilla's ContentProvider all the time, but i will change it so we only need one of them